library/src/main/java/org/jellyfin/apiclient/discovery/ServerDiscovery.kt

```diff
@@ -0,0 +1,100 @@
+package org.jellyfin.apiclient.discovery
+
+import com.google.gson.Gson
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.flow.flow
+import kotlinx.coroutines.isActive
+import org.jellyfin.apiclient.logging.ILogger
+import java.net.DatagramPacket
+import java.net.DatagramSocket
+import java.net.InetAddress
+
+class ServerDiscovery(
```

Comment:
I'm guessing it will be up to clients now to create an instance of this class and use it prior to initializing the `ConnectionManager`? Could we document the usage somewhere?

Reply:
I want to add a new class that is simply called `Jellyfin` which contains an instance of this class. The idea is to always use the `Jellyfin` class as entrypoint to the apiclient functionality.

A simple example will be something like this;
```kt
val jf = Jellyfin() // Optionally add options to the constructor
val flow = jf.serverDiscovery.discover(timeout = 30.seconds)

// And some future ideas:
val server = jf.server("https://demo.jellyfin.org/stable")
val info = server.getPublicInfo()

val user = server.user("token")
val user = server.authenticate("username", "password")
```

I'm still tinkering about that though, I want a solution that is compatible with OpenAPI generated code